pool.h

```diff
@@ -0,0 +1,44 @@
+/**
+ *  Copyright 2014-2015 MongoDB, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+#ifndef __MONGODB_DRIVER_POOL_H__
+#define __MONGODB_DRIVER_POOL_H__
+
+extern "C" {
+#include "../../../libmongoc/src/mongoc/mongoc.h"
+}
+
+namespace HPHP {
+
+class Pool
+{
+	public:
+		mongoc_client_pool_t *m_pool;
+
+		Pool(const mongoc_uri_t *uri)
+		{
+			m_pool = mongoc_client_pool_new(uri);
```

Comment:
I'm making assumptions here (as I'm not familiar with HHVM's innards), but if you're storing the pool map as a `thread_local`, that means that these pools are never going to be accessed from multiple request threads. You're just using the pool as a means to persist the topology between requests, and effectively using it in a single-threaded manner. If the pool is used in this way, each client is going to require 2 sockets per server due to the background monitoring thread created by the pool. Also, the various URI options for pools don't really apply, as we'd always want the pool as small as possible for such a use case.

Unless you actually intend to share the pool across all of HHVM, maybe it makes more sense to just keep a map of the clients themselves. That would maintain the singled-threaded logic in libmongoc (and SDAM) and avoid the extra monitoring thread for each pool.

If I've misunderstood `thread_local` and that pool map really is shared across the entire HHVM process, disregard what I'm saying here.